CHANGELOG.md

```diff
@@ -3,6 +3,31 @@
 All notable changes to this project will be documented in this file.
 This project adheres to [Semantic Versioning](http://semver.org/).
 
+## 4.0.0 - 2022-05-17
+
+### Changed
+
+-   Fixed a duplicate record name issue in offline-capable views if more than 1 form with the same title exist in the local database, closes (#417)
+
+    **Breaking change:** once users visit a form in offline-capable mode, the offline form cache IndexedDB schema will be upgraded, which is irreversible.
```

Comment:
What actually happens if someone downgrades? The form will still work but there will be no uniqueness constraint on name? Something worse?

Reply:
If they've opened a form in offline-capable mode before downgrading, `store.js` will upgrade the IndexedDB store to version 4. Opening the offline form again after downgrade will attempt to open the IndexedDB store as version 3. This will throw a `VersionError` and block loading the form.

I didn't know this at the time, but in hindsight this would also have been the case when we introduced last-saved. I suppose if we want to provide an escape hatch, we could also make a special release (tagged appropriately to describe its purpose) back-porting only the schema change without all of the other changes in this release.

Reply:
> this would also have been the case when we introduced last-saved

Oh! And we heard nothing of it so I assume no one ran into it, at least not at much scale.

> This will throw a VersionError and block loading the form.

Does the user-facing error message provide any insight into what to do next? Anything like contacting the site admin?

Reply:
> Does the user-facing error message provide any insight into what to do next? Anything like contacting the site admin?

This is the error I get:

![image](https://user-images.githubusercontent.com/199830/169117822-9a61d9b9-5002-41dc-9143-e04e76641684.png)

This definitely could use improvement, possibly even an option to wipe the DB.

Reply:
Could be improved but I think it's good enough for now since we don't think it's frequent.